pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -189,24 +189,114 @@ func (b *Build) BuildPod(images BuildPodImages, buildContext BuildContext) (*cor
 		MountPath: sourceVolume.MountPath,
 		SubPath:   b.Spec.Source.SubPath, // empty string is a nop
 	}
-
+	platformAPILessThan07 := platformAPI.LessThan(semver.MustParse("0.7"))
 	var genericCacheArgs []string
+	var analyzerCacheArgs []string = nil
 	var exporterCacheArgs []string
 	var cacheVolumes []corev1.VolumeMount
 
 	if b.Spec.NeedVolumeCache() && buildContext.os() != "windows" {
 		genericCacheArgs = []string{"-cache-dir=/cache"}
 		cacheVolumes = []corev1.VolumeMount{cacheVolume}
+		if platformAPILessThan07 {
+			analyzerCacheArgs = genericCacheArgs
+		}
 		exporterCacheArgs = genericCacheArgs
 	} else if b.Spec.NeedRegistryCache() {
 		useCacheFromLastBuild := (b.Spec.LastBuild != nil && b.Spec.LastBuild.Cache.Image != "")
 		if useCacheFromLastBuild {
 			genericCacheArgs = []string{fmt.Sprintf("-cache-image=%s", b.Spec.LastBuild.Cache.Image)}
 		}
+		analyzerCacheArgs = genericCacheArgs
 		exporterCacheArgs = []string{fmt.Sprintf("-cache-image=%s", b.Spec.Cache.Registry.Tag)}
 	} else {
 		genericCacheArgs = nil
 	}
+
+	analyzeContainer := corev1.Container{
+		Name:      "analyze",
+		Image:     b.Spec.Builder.Image,
+		Command:   []string{"/cnb/lifecycle/analyzer"},
+		Resources: b.Spec.Resources,
+		Args: args([]string{
+			"-layers=/layers",
+			"-analyzed=/layers/analyzed.toml"},
+			analyzerCacheArgs,
+			func() []string {
+				if platformAPILessThan07 {
+					return []string{
+						"-group=/layers/group.toml",
+					}
+				}
+				return []string{}
+			}(),
+			func() []string {
+				if platformAPILessThan07 {
+					return []string{}
+				}
+				tags := []string{}
+				for _, tag := range b.Spec.Tags {
+					tags = append(tags, "-tag="+tag)
+				}
+				return tags
+			}(),
+			func() []string {
+				if b.Spec.LastBuild != nil && b.Spec.LastBuild.Image != "" {
+					return []string{b.Spec.LastBuild.Image}
+				}
+				return []string{b.Tag()}
+			}(),
+		),
+		VolumeMounts: volumeMounts([]corev1.VolumeMount{
+			layersVolume,
+			workspaceVolume,
+			homeVolume,
+		}, func() []corev1.VolumeMount {
+			if platformAPILessThan07 {
+				return cacheVolumes
+			}
+			return []corev1.VolumeMount{}
+		}()),
+		Env: []corev1.EnvVar{
+			homeEnv,
+			{
+				Name:  platformAPIEnvVar,
+				Value: platformAPI.Original(),
+			},
+			serviceBindingRootEnv,
+		},
+		ImagePullPolicy: corev1.PullIfNotPresent,
+	}
+	analyzerContainerMods := ifWindows(
+		buildContext.os(),
+		addNetworkWaitLauncherVolume(),
+		useNetworkWaitLauncher(dnsProbeHost),
+		userprofileHomeEnv(),
+	)
+	detectContainer := corev1.Container{
+		Name:      "detect",
+		Image:     b.Spec.Builder.Image,
+		Command:   []string{"/cnb/lifecycle/detector"},
+		Resources: b.Spec.Resources,
+		Args: []string{
+			"-app=/workspace",
+			"-group=/layers/group.toml",
```

Comment:
It would be nice to be explicit about `/layers/analyzed.toml`.

Reply:
Not sure if I follow you? Detector doesn't take in analyzed as in input flag. It is specified for analyzer, restorer and exporter currently.